Add the possibility to build & install the Qt's demos and examples.
This patch is still to be evolved toward Qt full integration in Buildroot.

Signed-off-by: Jérémie Scheer <jeremie.scheer@armadeus.com>

Index: buildroot/package/qtopia4/Config.in
===================================================================
--- buildroot.orig/package/qtopia4/Config.in	2010-10-11 09:38:44.000000000 +0200
+++ buildroot/package/qtopia4/Config.in	2010-10-11 09:49:36.000000000 +0200
@@ -325,4 +325,14 @@
 	  Build the Qt Script Tools module.
 	  if unsure, say n.
 
+config BR2_PACKAGE_QT_DEMOS
+	bool "Demos"
+	help
+	  Build the Qt Demos.
+
+config BR2_PACKAGE_QT_EXAMPLES
+	bool "Examples"
+	help
+	  Build the Qt Examples.
+
 endif # BR2_PACKAGE_QT
Index: buildroot/package/qtopia4/qtopia4.mk
===================================================================
--- buildroot.orig/package/qtopia4/qtopia4.mk	2010-10-11 09:38:44.000000000 +0200
+++ buildroot/package/qtopia4/qtopia4.mk	2010-10-11 16:36:12.000000000 +0200
@@ -17,6 +17,7 @@
 QT_SITE:=http://get.qt.nokia.com/qt/source
 QT_CAT:=$(ZCAT)
 QT_TARGET_DIR:=$(BUILD_DIR)/qt-embedded-linux-opensource-src-$(QT_VERSION)
+QT_DEMOS_DIR:=/usr/local/qt-embedded-$(QT_VERSION)
 
 QT_CONFIGURE:=#empty
 
@@ -345,6 +346,14 @@
 QT_CONFIGURE+= -no-stl
 endif
 
+ifneq ($(BR2_PACKAGE_QT_DEMOS),y)
+QT_CONFIGURE+= -nomake demos
+endif
+
+ifneq ($(BR2_PACKAGE_QT_EXAMPLES),y)
+QT_CONFIGURE+= -nomake examples
+endif
+
 # ccache and precompiled headers don't play well together
 ifeq ($(BR2_CCACHE),y)
 QT_CONFIGURE += -no-pch
@@ -483,11 +492,11 @@
 		-no-accessibility \
 		-no-separate-debug-info \
 		-prefix /usr \
+		-examplesdir $(STAGING_DIR)/$(QT_DEMOS_DIR)-arm/examples \
+		-demosdir $(STAGING_DIR)/$(QT_DEMOS_DIR)-arm/demos \
 		-hostprefix $(STAGING_DIR)/usr \
 		-fast \
 		-no-rpath \
-		-nomake examples \
-		-nomake demos \
 	)
 	touch $@
 
@@ -560,8 +569,19 @@
 	cp -dpf $(STAGING_DIR)/usr/lib/libQtScriptTools.so.* $(TARGET_DIR)/usr/lib/
 endif
 
+$(TARGET_DIR)/$(QT_DEMOS_DIR)/demos/demos.pro:
+ifeq ($(BR2_PACKAGE_QT_DEMOS),y)
+	mkdir -p $(TARGET_DIR)/$(QT_DEMOS_DIR)/demos
+	cp -dpfR $(STAGING_DIR)/$(QT_DEMOS_DIR)-arm/demos/* $(TARGET_DIR)/$(QT_DEMOS_DIR)/demos/
+endif
+
+$(TARGET_DIR)/$(QT_DEMOS_DIR)/examples/examples.pro:
+ifeq ($(BR2_PACKAGE_QT_EXAMPLES),y)
+	mkdir -p $(TARGET_DIR)/$(QT_DEMOS_DIR)/examples
+	cp -dpfR $(STAGING_DIR)/$(QT_DEMOS_DIR)-arm/examples/* $(TARGET_DIR)/$(QT_DEMOS_DIR)/examples/
+endif
 
-$(TARGET_DIR)/usr/lib/libQtCore.so.4: $(STAGING_DIR)/usr/lib/libQtCore.la $(QT_LIBS)
+$(TARGET_DIR)/usr/lib/libQtCore.so.4: $(STAGING_DIR)/usr/lib/libQtCore.la $(QT_LIBS) $(TARGET_DIR)/$(QT_DEMOS_DIR)/demos/demos.pro $(TARGET_DIR)/$(QT_DEMOS_DIR)/examples/examples.pro
 	# Strip all installed libs
 ifeq ($(BR2_PACKAGE_QT_SHARED),y)
 	cp -dpf $(STAGING_DIR)/usr/lib/libQtCore.so.* $(TARGET_DIR)/usr/lib/
